doc/legacy_mode/main.md

```diff
@@ -34,36 +30,30 @@ The extension supports configuration of the tools:
 * [Rustfmt Configuration](rustfmt_configuration.md)
 * [Rustsym Configuration](rustsym_configuration.md)
 
-This mode supports configuration via the configuration parameters.
+You may also configure Legacy Mode via configuration parameters.
 
 ## Configuration Parameters
 
 ### Show Output
 
-The `"rust.showOutput"` configuration parameter controls whether the output channel should be shown when [a cargo command is started executing](../cargo_command_execution.md).
+The `"rust.showOutput"` configuration parameter controls whether the Output channel should be shown when [a Cargo command starts executing](../cargo_command_execution.md). All Cargo command other than `build` and `check` trigger this event.
 
 The possible values:
 
 * `true` - the output channel should be shown
 * `false` - the output channel shouldn't be shown
 
-### Execute Cargo command in a terminal
-
-The `"rust.executeCargoCommandInTerminal"` configuration parameter controls whether [a cargo command should be executed](../cargo_command_execution.md) in an integrated terminal.
-
-By default, the extension executes a cargo command as a child process.
-
-Then it parses the output of the cargo command and publishes diagnostics.
+When `"rust.executeCargoCommandInTerminal"` is set to `true`, `"rust.showOutput"` has no effect.
 
-However, it may be changed.
+### Executing Cargo commands in the integrated temrinal
 
-It is useful if you need to run a binary and enter some text.
+The `"rust.executeCargoCommandInTerminal"` configuration parameter controls whether [a Cargo command should be executed](../cargo_command_execution.md) in an integrated terminal.
 
-Unfortunately, there is no way to parse output of an integrated terminal.
+By default, the extension executes a Cargo command as a child process. It then parses the output of the command and publishes diagnostics. This is useful if you need to run a binary and enter some text.

```

Comment:
Replace **This is useful** with **Executing Cargo command in an integrated terminal is useful**.
Why do you write **a Cargo**? I thought the correct variant was "Cargo" because it is the tool's name.